comms/core/src/lib.rs

```diff
@@ -64,3 +64,10 @@ pub use async_trait::async_trait;
 pub use bytes::{Buf, BufMut, Bytes, BytesMut};
 #[cfg(feature = "rpc")]
 pub use tower::make::MakeService;
+
+/// Was the connection closed due to minimize_connections
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Minimized {
+    Yes,
+    No,
+}
```

Comment:
It appears that this is only used to improve logs.

Not necessary for now but I'd suggest making this a general reason enum (you can use thiserror for the human readable text or impl Display)

```suggestion
#[derive(Debug, Clone, Copy, thiserror::Error)]
pub enum DisconnectReason {
    #[error("Remote disconnected")]
    RemoteDisconnected,
    #[error("Connection was culled because it was inactive")]
    ConnectionCulledInactive, 
     #[error("Connection was culled to reduce number of connections")]
    ConnectionCulledMinimise
    #[error("Local node explicit disconnect")]
    LocalDisconnected
}
```

Reply:
Issue created for this: #6331